examples/virtual_methods/purrable.rs

```diff
@@ -0,0 +1,115 @@
+use glib::object::InterfaceRef;
+use glib::prelude::*;
+use glib::subclass::prelude::*;
+
+mod iface {
+    use super::*;
+
+    /// Custom class struct for the [`super::Purrable`] interface.
+    ///
+    /// The class struct is used to implement the vtable for method dispatch. The first field *must*
+    /// be a pointer to the parent type. In our case, this is `GTypeInterface`
+    #[derive(Copy, Clone)]
+    #[repr(C)]
+    pub struct Purrable {
```

Comment:
This should be called `PurrableInterface`, just like the corresponding other one is called `PetClass`

Reply:
Like @pbor mentioned, should also get a `#[repr(C)] pub struct Purrable(std::ffi::c_void);` for the instance pointers

Reply:
Or could call them just `Class`, `Instance`, `Interface` etc because we get namespacing from the modules anyway

Reply:
They would be used whenever you have FFI code calling into your interface. Here it's not necessary and I don't think you would actually encounter them without exposing a C API, but probably a good idea to have them here for consistency?

That's basically the equivalent of `gio::ffi::GInitable`

Reply:
Maybe this also tells us that these `repr(C)` structs should actually be in an `ffi` module? Not sure

Reply:
Question is whether the `Class` / `Interface` structs should also be in that module, I guess

Reply:
Also for `Pet` as an abstract class, you would use the `ObjectSubclass::Instance` struct